auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -679,6 +702,51 @@ public AuthenticationConfiguration usePassword(String password) {
         return usePassword(password == null ? null : ClearPassword.createRaw(ClearPassword.ALGORITHM_CLEAR, password.toCharArray()));
     }
 
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given masked password to authenticate.
+     *
+     * @param maskedPasswordBytes the masked password bytes (must not be {@code null})
+     * @param algorithm the algorithm (can be {@code null})
+     * @param initialKeyMaterial the initial key material (can be {@code null})
+     * @param iterationCount the iteration count (must not be 0)
+     * @param salt the salt bytes (must not be {@code null})
+     * @param initializationVector the initialization vector (can be {@code null})
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration usePassword(byte[] maskedPasswordBytes, String algorithm, char[] initialKeyMaterial, int iterationCount, byte[] salt, byte[] initializationVector) throws Exception {
```

Comment:
I think it would be good to call this ```useMaskedPassword``` as well just to be consistent with the above method.

Reply:
fixed